code/Controllers/CMSMain.php

```diff
@@ -1949,12 +1933,9 @@ public function revert($data, $form)
      *
      * @see deletefromlive()
      *
-     * @param array $data
-     * @param Form $form
-     * @return HTTPResponse
      * @throws HTTPResponse_Exception
      */
-    public function delete($data, $form)
+    public function delete(array $data, Form $form): HTTPResponse
```

Comment:
This method sometimes returns `Security::permissionFailure()` which sometimes returns an array

Reply:
As noted above, it does not return an array